### ISSUE TITLE
categorial array not stored in .uns in v0.7rc1

### ISSUE THREAD

**Issue by picciama** [theislab/anndata#292]
I am currently writing an interface for diffxpy/batchglm and MASS/MAST/edgeR since we are trying to compare the fitting/DE tools between available R packages and diffxpy. 
v0.6.22 needed to be updated since it couldn't read in h5ad files any longer (probably due to the pandas update)

`AnnData.write` in v0.6.22 stored the levels of categorical variables in `.uns`. This had the advantage that reading the file with R (using `rhdf5` library) was easily possible, since it recognized `$var` and `$obs` as `data.frame` object. 
Since 0.7rc1, it is stored differently, so R now recognizes it as a list. 
Here an example, with a condition and batch column in the obs dataframe after reading it in R:

Old structure (v0.6.22):
```
X$obs: data.frame with names(X$obs) = c('condition', 'column')

X$uns: $batch_categories
       $condition_categories
```
This is a data.frame and a list.

New structure (v0.7rc1):
```
X$obs: $`__categories`: $condition
                        $batch
       $condition
       $batch
```
This is a list.

Can you already tell me what the structure will be when 0.7 is officially released?

This is related to #171 I think (see comment https://github.com/theislab/anndata/issues/171#issuecomment-569735914).

**Comment by ivirshup** [comment]
We've moved to a new on-disk representation for dataframes which clears up some of the problems with how we represented categorical values (for example, if `species` was a categorical in `obs` and `var`, only one of those would be stored in `uns`). The documentation for this is on this #267, in particular [here](https://github.com/theislab/anndata/blob/87707a26c1a403c0aac8f0b9978582f3bdb647b0/docs/fileformat-prose.rst#dataframes). Hope this helps!
